src/native/corehost/corehost.cpp

```diff
@@ -72,10 +72,10 @@ bool is_exe_enabled_for_execution(pal::string_t* app_dll)
         return false;
     }
 
-    std::string binding(&embed[0]);
+    size_t binding_len = strlen(&embed[0]);
```

Comment:
nit: C arrays convert to pointers to their first element. This can be: `strlen(embed)`.

For the `compare_memory_nooptimization`, I suggested to use `&embed[0]` for consistency with `&embed[hi_len]`.

Reply:
I'm aware. I'm trying to keep access consistent in this case because I don't know when or where any of these optimizations will trigger.

Reply:
Related to this.

The comment higher up should be updated:
```
// Must not be 'const' because std::string(&embed[0]) below would bind to a const string ctor plus length
// where length is determined at compile time (=64) instead of the actual length of the string at runtime.
```
to something like:
```
// Must not be 'const' because strlen below would be determined at compile time (=64) instead of the actual length of the string at runtime.
```

This is what avoids optimizations here.